compiler/z/codegen/S390Peephole.cpp

```diff
@@ -1349,26 +1349,66 @@ TR_S390PostRAPeephole::ConditionalBranchReduction(TR::InstOpCode::Mnemonic branc
 
 
 /**
- * Catch the pattern where an CR/BRC can be converted
- *    CR R1,R2
- *    BRC  Mask, Lable
+ * Catch the pattern where an CLR/BRC can be converted
+ *    CLR R1, R2
+ *    BRC Mask, Lable
  * Can be replaced with
- *    CRJ  R1,R2,Lable
+ *    CLRJ R1, R2, Lable, Mask
  */
 bool
 TR_S390PostRAPeephole::CompareAndBranchReduction()
    {
-   bool performed = false;
+   if (!TR::Compiler->target.cpu.getSupportsArch(TR::CPU::z10))
+      return false;
 
-   TR::Instruction *current = _cursor;
-   TR::Instruction *next = _cursor->getNext();
-   TR::InstOpCode::Mnemonic curOpCode = current->getOpCodeValue();
-   TR::InstOpCode::Mnemonic nextOpCode = next->getOpCodeValue();
-   if ((curOpCode == TR::InstOpCode::CR || curOpCode == TR::InstOpCode::CGR || curOpCode == TR::InstOpCode::CGFR)
-      && (nextOpCode == TR::InstOpCode::BRC || nextOpCode == TR::InstOpCode::BRCL))
-      {
-      printf("Finding CR + BRC\n");
-      printf("method=%s\n", comp()->signature());
+   bool performed = false;
+   TR::Compilation *comp = _cg->comp();
+   TR::S390RRInstruction *clrInstruction = (TR::S390RRInstruction*)_cursor;
+   TR::S390BranchInstruction *brcInstruction = (TR::S390BranchInstruction*)_cursor->getNext();
```

Comment:
Prefer to use C++ style type-safe casts (`static_cast`) instead of C-style unsafe casts. The compiler will catch obvious issues both during development and future refactoring.

However this code is not safe to begin with because the `S390RRInstruction` and `S390BranchInstruction` may override the `getOpCodeValue()` API call. The cast should only happen if and only if we have a `CLR` and `BRC` sequence, so these declarations need to move into the `if` block below.